### PR TITLE
Completed Pre-Course 1

### PR DESCRIPTION
**Pull request by Tirthaj**

### CODE REVIEW

**Comment by Tirthaj** [comment]
Rectified exercise 1

File: Exercise_1.py [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+            self.data.append(e)
+            self.top +=1
+            print('Push operation Successful')
+            return
+        else:
+            print('Stack Overflow')
+            return
 
     def pop(self):
         if self.isempty() == 0:
```
Comment (yahska-ednihs) [inline Exercise_1.py:22]: Use top variable here to determine if stack is empty

File: Exercise_2.py [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+        self.data = data
+        self.next = None
+
 class Stack:
     def __init__(self):
-        
-    def push(self, data):
-        
+        self.head = None
+
```
Comment (yahska-ednihs) [inline Exercise_2.py:9]: Good Job! Please add comments and time/space complexity for each operation

File: Exercise_3.py [hunk 3]
```diff
@@ -13,20 +15,79 @@ def __init__(self):
...
+    def display(self):
+        if self.head == None:
+            print('LL is empty cant display')
+            return
+        else:
+            n = self.head
+            while n != None:
+                print(n.data)
+                n = n.next
+LL = SinglyLinkedList()
```
Comment (yahska-ednihs) [inline Exercise_3.py:87]: Good Job with implementation, comments and time complexities. Please try to add space complexity.

**Pull request closed without merging**